app/models/flag.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class Flag < ActiveRecord::Base
+  scope :enabled, -> { where(enabled: true) }
+  scope :system, -> { where("id < 1000") }
+
+  before_save :set_position
+  before_save :set_name_key
+  after_save :reset_flag_settings!
+  after_destroy :reset_flag_settings!
+
+  default_scope { order(:position) }
+
+  def used?
+    PostAction.exists?(post_action_type_id: self.id) ||
+      ReviewableScore.exists?(reviewable_score_type: self.id)
+  end
+
+  def self.reset_flag_settings!
```

Comment:
I think on this we could have a brief comment about why `DiscourseEvent` is used (for the multiprocess thing) so people don't think it's weird.